src/components/common/state-machine/state-machine.ts

```diff
@@ -738,6 +747,15 @@ const authStateMachine = createMachine(
       requiresMFAAuthAppCode: (context) =>
         context.mfaMethodType === MFA_METHOD_TYPE.AUTH_APP &&
         context.requiresTwoFactorAuth === true,
+      requiresMFASmsCode: (context) =>
```

Comment:
forget to clean this up?

Reply:
No, I left this up on purpose, the requiresMFAAuthAppCode is a gate that's used for the login journey, I created 2 new gates for the Reset Password flow, that ensure that the user has mfa set up, that it is a password reset journey and that the switch is turned on. This was the approach I came up with to be entirely sure that the code is clear and that it works well.  I tried other approaches but for them there were integration tests and unit tests that were failing. I am open to suggestions for any better approach though.